### DISCUSSION TITLE
SSR with Docusaurus (SSG)

### DISCUSSION THREAD

**Discussion by tibotiber** [mantinedev/mantine#317] [Q&A] [answered]
Hi, thanks for the great work on Mantine. I'm having an issue where `MantineProvider` doesn't seem to work in SSR. I have seen that you have docs for SSR [if you run the server](https://mantine.dev/theming/ssr/), or in Jamstack use-cases with [Next](https://mantine.dev/theming/next/) and [Gatsby](https://mantine.dev/theming/gatsby/). What about other static site generators? I am using [Docusaurus](https://docusaurus.io) in this particular case. 

Would you be able to explain the conceptual steps to take such that someone might be able to write a plugin?

Thanks in advance for any help you may be able to provide.

**Comment by tibotiber** [comment]
FWIW, my current ugly workaround is to force a render once the page has loaded on the client.

```jsx
function Page({ children }) {
  const { isDarkTheme } = useThemeContext();

  // ugly workaround mantine SSR issue
  const [forceLightTheme, setForceLightTheme] = useState(false)
  useEffect(() => {
    setForceLightTheme(false)
  }, [])

  return (
    <MantineProvider
      theme={{
        colorScheme: (!forceLightTheme && isDarkTheme) ? 'dark' : 'light', // sync docusaurus and mantine themes
        fontFamily: '"RNSSanz-Normal", Helvetica, sans-serif',
        primaryColor: 'dark',
        black: '#3a3c3c',
        colors: {
          dark: ['#C1C2C5', '#A6A7AB', '#909296', '#5C5F66', '#373A40', '#2a2c2c', '#242526', '#1A1c1c', '#141517', '#101113'], // replace dark[5-7] only
          gray: ['#f7f9fb', '#F1F3F5', '#E9ECEF', '#DEE2E6', '#CED4DA', '#ADB5BD', '#868E96', '#495057', '#343A40', '#212529'] // replace gray[0] only
        }
      }}
    >
      <main className="container container--fluid margin-vert--lg">
        {children}
      </main>
    </MantineProvider>
  )
}
```

The issue with that is I'm still getting a layout jump if the page was server rendered.

**Comment by rtivital** [comment, marked as answer]
Hi @tibotiber you can find ssr docs here – https://mantine.dev/theming/ssr/

General approach:
- render app to string
- render `ServerStyles` component in head with page html content

**Comment by tibotiber** [comment]
Thanks for the answer @rtivital. I've looked into writing a plugin on the Docusaurus side but it seems they do not support it yet – https://github.com/facebook/docusaurus/issues/3236. Thought this might be useful for anyone else looking for this.

In the meantime, my workaround is fine if users can bear with flash of unstyled content.

**Reply by zac-az** [reply]
Hey @tibotiber, did you manage to find a way to solve this? The page workaround can only work for pages, and docusauras is made up of mostly docs and blogs. The pages is only one part of this

The SSR completely breaks MantineProvider and I can't see how to integrate the two tools at the moment.

**Reply by tibotiber** [reply]
Still on my workaround. Admittedly I haven't dug much further. I'm only using Mantine in pages so no issue for me. Sorry.

**Reply by zac-az** [reply]
Okay thanks @tibotiber . Please see https://github.com/mantinedev/mantine/discussions/317#discussioncomment-5802958 for additional info :)

**Reply by rcronin** [reply]
@tibotiber - do you have a repo available to show your full workaround by chance?

**Reply by tibotiber** [reply]
yes, have a look at https://github.com/smplrspace/docs/blob/main/src/components/Page.js#L10

**Comment by zac-az** [comment]
Please also see https://github.com/facebook/docusaurus/discussions/8959#discussioncomment-5802859 for a detailed explanation from Docusauras.

I believe Mantine moving to pure css will be a welcome addition and should allow the integration to work properly. It would also be very much inline with my preferences, css variables will be a breeze.
